### ISSUE TITLE
Not clear where to add @mustache annotation

### ISSUE THREAD

**Issue by MikeMitterer** [xxgreg/mustache#18]
I really like your package - and it works great in Chromium but after compiling the whole App to JS I get an error message:
 
I have this class:
```dart
@mustache
class Model {
    static Model _model;

    final StreamController _controller = new StreamController<ModelChangedEvent>.broadcast();

    Stream<ModelChangedEvent> onChange;

    int _sliderValue = 20;

    @mustache
    List<int> randomValues = new List<int>();

    factory Model() {
        if(_model == null) {  _model = new Model._internal(); }
        return _model;
    }

    @mustache
    int get sliderValue => _sliderValue;

    set sliderValue(final int value) {
        _sliderValue = value;
        randomValues.clear();
        for(int counter = 0;counter < _sliderValue;counter++) {
            randomValues.add(new Math.Random().nextInt(1000));
        }
        _controller.add(new ModelChangedEvent());
    }

    //- private -----------------------------------------------------------------------------------

    Model._internal() {
        onChange = _controller.stream;
    }
}
```
After compiling to JS I get the following error:
```
Uncaught Uncaught Error: Value was missing for variable tag: sliderValue. (1:18)
<div>Slidervalue:{{sliderValue}}<ol>{{#randomValues}}<li>{{.}},</li>{{/rand...
                 ^

Stack Trace:
Value was missing for variable tag: sliderValue. (1:18)
<div>Slidervalue:{{sliderValue}}<ol>{{#randomValues}}<li>{{.}},</li>{{/rand...
```

**Comment by MikeMitterer** [comment]
Sry - forgot to mention that I have enables mirrors...
```yaml
transformers:
- $dart2js:
    commandLineOptions: [--enable-experimental-mirrors]
```

**Comment by xxgreg** [comment]
From memory you should only have to add it to the class definition, not the individual members.

For some background info on how this works:

The @mustache annotation is a MirrorsUsed metaTarget.

For example this is how it is defined:

```dart
@MirrorsUsed(metaTargets: const [m.mustache])
import 'dart:mirrors';

...

const MustacheMirrorsUsedAnnotation mustache = const MustacheMirrorsUsedAnnotation();

class MustacheMirrorsUsedAnnotation {
  const MustacheMirrorsUsedAnnotation();
}
```

**Comment by xxgreg** [comment]
Ok. I had a look at this. It is broken - sorry about that :(

I'm hoping this will fix it:
https://github.com/xxgreg/mustache/commit/b0e8e59a4a2ef16d9518e1bfa5b776c15d0a02dc

Can you change your pubspec.yaml to use a git dependency and see if it fixes your problem. If so I'll push a new version to pub later today.

```
dependencies:
  mustache:
    git: git://github.com/xxgreg/mustache
```

**Comment by xxgreg** [comment]
And to answer your first question properly - you can use the @mustache annotation either on the class declaration, which will include all members (this includes inherited ones too I think), or you can just put the annotation against the individual members which you want to use in tags. There is no need to do both.

**Comment by MikeMitterer** [comment]
Cool man - Thanks for help! Works like a charm now. Can you publish it to pub?

**Comment by xxgreg** [comment]
Published 0.2.2.

Thanks for the filing the bug, I haven't been using this on the client side, only the server. So it is very useful to get feedback like this.

Cheers ;)
